core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -460,8 +460,104 @@ func (network *DefaultServiceNetwork) AddServices(
+// way to do soft-delete for containers. Once it's deleted, it's gone, so if Kurtosis fails at re-creating it, it
+// doesn't rollback to the state previous to calling this function
+func (network *DefaultServiceNetwork) UpdateServices(ctx context.Context, updateServiceConfigs map[service.ServiceName]*service.ServiceConfig, batchSize int) (map[service.ServiceName]*service.Service, map[service.ServiceName]error, error) {
+	network.mutex.Lock()
+	defer network.mutex.Unlock()
+	failedServicesPool := map[service.ServiceName]error{}
+	successfullyUpdatedService := map[service.ServiceName]*service.Service{}
+
+	// First, remove the service
+	serviceUuidToNameMap := map[service.ServiceUUID]service.ServiceName{}
+	serviceUuidsToRemove := map[service.ServiceUUID]bool{}
+	for serviceName := range updateServiceConfigs {
+		serviceRegistration, found := network.registeredServiceInfo[serviceName]
+		if !found {
+			failedServicesPool[serviceName] = stacktrace.NewError("Unable to update service that is not registered "+
+				"inside this enclave: '%s'", serviceName)
+		} else {
+			serviceUuid := serviceRegistration.GetUUID()
+			serviceUuidsToRemove[serviceUuid] = true
+			serviceUuidToNameMap[serviceUuid] = serviceName
+		}
+	}
+	successfullyRemovedServices, failedRemovedServices, err := network.kurtosisBackend.RemoveRegisteredUserServiceProcesses(ctx, network.enclaveUuid, serviceUuidsToRemove)
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "Unexpected error happened updating services")
+	}
+	for serviceUuid, serviceErr := range failedRemovedServices {
+		if serviceName, found := serviceUuidToNameMap[serviceUuid]; found {
+			failedServicesPool[serviceName] = serviceErr
+		} else {
+			return nil, nil, stacktrace.NewError("Error mapping service UUID to service name. This is a bug in Kurtosis")
+		}
+	}
+
+	// Set service status back to registered and remove its currently saved service config
+	for serviceUuid := range successfullyRemovedServices {
+		if serviceName, found := serviceUuidToNameMap[serviceUuid]; found {
+			if network.isPartitioningEnabled {
+				if sidecar, found := network.networkingSidecars[serviceName]; found {
+					if err = network.networkingSidecarManager.Remove(ctx, sidecar); err != nil {
+						failedServicesPool[serviceName] = stacktrace.Propagate(err, "An error occurred destroying the sidecar for service with name '%v'", serviceName)
+					} else {
+						delete(network.networkingSidecars, serviceName)
+					}
+				}
+			}
+
+			network.registeredServiceInfo[serviceName].SetStatus(service.ServiceStatus_Registered)
+			network.registeredServiceInfo[serviceName].SetConfig(nil)
+		} else {
+			return nil, nil, stacktrace.NewError("Error mapping service UUID to service name. This is a bug in Kurtosis")
+		}
+	}
+
+	// Re-create service with the new service config
+	serviceToRecreate := map[service.ServiceUUID]*service.ServiceConfig{}
+	for serviceUuid := range successfullyRemovedServices {
```

Comment:
Some items from `successfullyRemovedServices` could have failed in the previous step, so I think, not 100% sure, it should iterate the result of `successfullyRemovedServices - serviceFailedInThePreviousStep` instead of the entire `successfullyRemovedServices` but, should be the case that it doesn't matter because bot operations are independent ??

Reply:
Nope you're totally right, I'll update